core/adapters/json_parse.go

```diff
@@ -52,26 +52,27 @@ func (jpa *JSONParse) Perform(input models.RunInput, _ *store.Store) models.RunO
 		return models.NewRunOutputError(err)
 	}
 
-	js, err := simplejson.NewJson([]byte(val))
-	if err != nil {
-		return models.NewRunOutputError(err)
+	if !gjson.Valid(val) {
+		return models.NewRunOutputError(errors.New("invalid json input"))
 	}
+	js := gjson.Parse(val)
```

Comment:
Even though `input.Result()` is of type gjson.Result, the input is sometimes encoded twice - that's certainly the case in the tests. I have taken the safer route to parse the input again using gjson. It's not very nice, I susspect it is related to these lines https://github.com/smartcontractkit/chainlink/pull/3038/files#diff-4cf19e75aa397a77b9d2d738a7d184c4R44-R49

Reply:
Yeah this part of the code is a bit of a nightmare.